File: nim/BinarySearch.nim
type
    arr = array[5,int]
let arr1 : arr = [1,2,3,4,5]

proc BinarySearch(arr1:arr,k:int): bool =
    var L = 0
    var r = 4
    while L < r:
        let tmp = (L+r) div 2
        if arr1[tmp] == k:
            return true
        if arr1[tmp] > k:
            r = tmp - 1
        else:
            L = tmp + 1 
    return false
var r = BinarySearch(arr1,10)
echo r

File: java/BinarySearch.java
class BinarySearch{
	public static void main(String[] args){
		int[] arr = {1,2,3,4,5};
		boolean r = find(arr,3);
		System.out.println(r);
	}
	public static boolean find(int[] arr,int k){
		int l = 0;
		int r = arr.length;
		while(l < r){
			int m = (l+r)/2;
			if(arr[m] == k) return true;
			if(arr[m] > k) r = m - 1;
			else l = m + 1;

		}
		return false;
	}
}
